lib/ukallocbbuddy/bbuddy.c

```diff
@@ -173,8 +182,14 @@ static void map_alloc(struct uk_bbpalloc *b, uintptr_t first_page,
 			memr->mm_alloc_bitmap[curr_idx] = ~0UL;
 		memr->mm_alloc_bitmap[curr_idx] |= (1UL << end_off) - 1;
 	}
-
+	
+#ifdef CONFIG_LIBUKALLOCBBUDDY_SMP_SAFE
+	uk_spin_lock(&b->global_lock);
```

Comment:
you should disable the interrupts before acquiring the lock and reenable them after releasing the lock.
```
irqf = ukplat_lcpu_save_irqf();
uk_spin_lock();
ukplat_lcpu_restore_irqf(irqf);
```
`

Reply:
Do we expect to have buddy allocator calls from interrupt handlers?